
## Pseudogenes and intronless

### Pseudogenes

Parsing encode gtf and extract gene names, gene ids and gene types (pseudogenes ...)

```python
#cd /projects/qbio/bifo/users/sergio/repository/20201122_hush
#mkdir annotation && cd annotation
#python

ifile = open("/projects/qbio/bifo/users/sergio/reference/gencode/release35_GRCh38.p13/gtf/gencode.v35.annotation.gtf")
ilines = ifile.readlines()
ifile.close()

genes = []

for l in ilines:
  if not l.startswith("##"):
    fields = l.split("\t")
    if fields[2] == "gene":
      chr = fields[0]
      start = fields[3]
      end = fields[4]
      strand = fields[6]
      annotations = fields[8].split("; ")[:-1]
      for a in annotations:
        if a.startswith("gene_id"):
          gene_id=a.split('"')[1]
        if a.startswith("gene_type"):
          gene_type=a.split('"')[1]
        if a.startswith("gene_name"):
          gene_name=a.split('"')[1]
      genes.append((chr, start, end, gene_name, gene_id, strand, gene_type))


ofile = open("gencode.v35.annotation.gene_type.bed", "w")
ofile.write("\n".join(["\t".join(g) for g in genes]) + "\n")
ofile.close()
```



### Intronless

As requested in [biostars](https://www.biostars.org/p/472156/)

```bash
cd /projects/qbio/bifo/users/sergio/repository/20201122_hush/annotation

awk '{ if ($3=="exon") print $1, $4, $5, $10 }' /projects/qbio/bifo/users/sergio/reference/gencode/release35_GRCh38.p13/gtf/gencode.v35.annotation.gtf | \
sort -u | \
cut -d'"' -f2 | \
sort | uniq -c | \
perl -lane '$F[0] == 1 and print $F[1]' > gencode.v35.annotation.intronless.txt

wc -l gencode.v35.annotation.intronless.txt # 22499
```



### Merge

```python
#cd /projects/qbio/bifo/users/sergio/repository/20201122_hush/annotation
#python

ifile = open("gencode.v35.annotation.gene_type.bed")
ilines = ifile.readlines()
ifile.close()

ifile2 = open("gencode.v35.annotation.intronless.txt")
ilines2 = [l.replace("\n", "") for l in ifile2.readlines()]
ifile2.close()

genes = []

for l in ilines:
  fields=l.split()
  gene_id=fields[4]
  if gene_id in ilines2:
    intronless = "y"
  else:
    intronless = "n"
  fields.append(intronless)
  genes.append(fields)


ofile = open("gencode.v35.annotation.gene_type.intronless.bed", "w")
ofile.write("\n".join(["\t".join(g) for g in genes]) + "\n")
ofile.close()
```

Check gene types and intronless:

```bash
cd /projects/qbio/bifo/users/sergio/repository/20201122_hush/annotation

wc -l gencode.v35.annotation.gene_type.intronless.bed # 60656

head gencode.v35.annotation.gene_type.intronless.bed | column -t
# chr1  11869  14409   DDX11L1      ENSG00000223972.5  +  transcribed_unprocessed_pseudogene  n
# chr1  14404  29570   WASH7P       ENSG00000227232.5  -  unprocessed_pseudogene              n
# chr1  17369  17436   MIR6859-1    ENSG00000278267.1  -  miRNA                               y
# chr1  29554  31109   MIR1302-2HG  ENSG00000243485.5  +  lncRNA                              n
# chr1  30366  30503   MIR1302-2    ENSG00000284332.1  +  miRNA                               y
# chr1  34554  36081   FAM138A      ENSG00000237613.2  -  lncRNA                              n
# chr1  52473  53312   OR4G4P       ENSG00000268020.3  +  unprocessed_pseudogene              y
# chr1  57598  64116   OR4G11P      ENSG00000240361.2  +  transcribed_unprocessed_pseudogene  n
# chr1  65419  71585   OR4F5        ENSG00000186092.6  +  protein_coding                      n
# chr1  89295  133723  AL627309.1   ENSG00000238009.6  -  lncRNA                              n

cut -f7 gencode.v35.annotation.gene_type.intronless.bed | sort | uniq -c | sort -k1,1nr
  # 19954 protein_coding
  # 16899 lncRNA
  # 10169 processed_pseudogene
  #  2615 unprocessed_pseudogene
  #  2212 misc_RNA
  #  1901 snRNA
  #  1881 miRNA
  #  1058 TEC
  #   943 snoRNA
  #   941 transcribed_unprocessed_pseudogene
  #   500 transcribed_processed_pseudogene
  #   497 rRNA_pseudogene
  #   188 IG_V_pseudogene
  #   144 IG_V_gene
  #   138 transcribed_unitary_pseudogene
  #   106 TR_V_gene
  #    97 unitary_pseudogene
  #    79 TR_J_gene
  #    49 polymorphic_pseudogene
  #    49 scaRNA
  #    47 rRNA
  #    37 IG_D_gene
  #    33 TR_V_pseudogene
  #    22 Mt_tRNA
  #    18 IG_J_gene
  #    18 pseudogene
  #    14 IG_C_gene
  #     9 IG_C_pseudogene
  #     8 ribozyme
  #     6 TR_C_gene
  #     5 sRNA
  #     4 TR_D_gene
  #     4 TR_J_pseudogene
  #     3 IG_J_pseudogene
  #     2 Mt_rRNA
  #     2 translated_processed_pseudogene
  #     1 IG_pseudogene
  #     1 scRNA
  #     1 translated_unprocessed_pseudogene
  #     1 vault_RNA

cut -f8 gencode.v35.annotation.gene_type.intronless.bed | sort | uniq -c
  # 38157 n
  # 22499 y
```





## RNA-seq

K562 datasets:

GSE | GSM name | SRX | SRR
----|----------|-----|-----
GSE95374 | GSM2509511	RNA-seq_WT_rep1 | SRX2589983 | SRR5287124
GSE95374 | GSM2509512	RNA-seq_WT_rep2 | SRX2589984 | SRR5287125
GSE95374 | GSM2509513	RNA-seq_MORC2-KO_rep1 | SRX2589985 | SRR5287126
GSE95374 | GSM2509514	RNA-seq_MORC2-KO_rep2 | SRX2589986 | SRR5287127
GSE95374 | GSM2509515	RNA-seq_MPP8-KO_rep1 | SRX2589987 | SRR5287128
GSE95374 | GSM2509516	RNA-seq_MPP8-KO_rep2 | SRX2589988 | SRR5287129
GSE95374 | GSM2509517	RNA-seq_TASOR-KO_rep1 | SRX2589989 | SRR5287130
GSE95374 | GSM2509518	RNA-seq_TASOR-KO_rep2 | SRX2589990 | SRR5287131



### Download fastq files

```bash
cd /projects/qbio/bifo/users/sergio/repository/20201122_hush

mkdir -p rnaseq/fastq && cd rnaseq/fastq

for run_id in SRR5287124 SRR5287125 SRR5287126 SRR5287127 SRR5287128 SRR5287129 SRR5287130 SRR5287131
do
  sbatch -J $run_id -o $run_id.log --mem 16G -t 1-0 --wrap "fastq-dump --outdir . --gzip --split-files -F $run_id"
done

tail *.log
```



### Rename fastq files

```bash
cd /projects/qbio/bifo/users/sergio/repository/20201122_hush/rnaseq/fastq

# GSE95374 | GSM2509511	RNA-seq_WT_rep1 | SRX2589983 | SRR5287124
for filename in SRR5287124_{1,2}.fastq.gz; do mv "$filename" "WT_rep1_$filename"; done;
# GSE95374 | GSM2509512	RNA-seq_WT_rep2 | SRX2589984 | SRR5287125
for filename in SRR5287125_{1,2}.fastq.gz; do mv "$filename" "WT_rep2_$filename"; done;
# GSE95374 | GSM2509513	RNA-seq_MORC2-KO_rep1 | SRX2589985 | SRR5287126
for filename in SRR5287126_{1,2}.fastq.gz; do mv "$filename" "MORC2-KO_rep1_$filename"; done;
# GSE95374 | GSM2509514	RNA-seq_MORC2-KO_rep2 | SRX2589986 | SRR5287127
for filename in SRR5287127_{1,2}.fastq.gz; do mv "$filename" "MORC2-KO_rep2_$filename"; done;
# GSE95374 | GSM2509515	RNA-seq_MPP8-KO_rep1 | SRX2589987 | SRR5287128
for filename in SRR5287128_{1,2}.fastq.gz; do mv "$filename" "MPP8-KO_rep1_$filename"; done;
# GSE95374 | GSM2509516	RNA-seq_MPP8-KO_rep2 | SRX2589988 | SRR5287129
for filename in SRR5287129_{1,2}.fastq.gz; do mv "$filename" "MPP8-KO_rep2_$filename"; done;
# GSE95374 | GSM2509517	RNA-seq_TASOR-KO_rep1 | SRX2589989 | SRR5287130
for filename in SRR5287130_{1,2}.fastq.gz; do mv "$filename" "TASOR-KO_rep1_$filename"; done;
# GSE95374 | GSM2509518	RNA-seq_TASOR-KO_rep2 | SRX2589990 | SRR5287131
for filename in SRR5287131_{1,2}.fastq.gz; do mv "$filename" "TASOR-KO_rep2_$filename"; done;
```



### Trim illumina adapters and quality trimming

```bash
cd /projects/qbio/bifo/users/sergio/repository/20201122_hush/rnaseq/fastq

mkdir ../fastq_trimmed

for fq1 in *_1.fastq.gz
do
  fq2=${fq1/_1./_2.}
  bname=${fq1%_1.fastq.gz}
  #echo $fq1, $fq2, $bname
  sbatch -J $bname -o ../fastq_trimmed/$bname.log --mem 4G -t 1-0 --wrap "cutadapt -a AGATCGGAAGAGC -A AGATCGGAAGAGC -m 15 -q 20 -o ../fastq_trimmed/$fq1 -p ../fastq_trimmed/$fq2 $fq1 $fq2 > ../fastq_trimmed/$bname.txt"
done

cd ../fastq_trimmed

tail *.log

grep "Read 1 with adapter" *.txt
grep "Read 2 with adapter" *.txt
grep "Pairs written (passing filters):" *.txt
```



### Alignment (rsem)

#### Prepare reference

```bash
cd /projects/qbio/bifo/users/sergio/reference/gencode/release35_GRCh38.p13

mkdir rsem_bowtie2

rsem-prepare-reference \
--gtf /projects/qbio/bifo/users/sergio/reference/gencode/release35_GRCh38.p13/gtf/gencode.v35.annotation.gtf \
--bowtie2 \
/projects/qbio/bifo/users/sergio/reference/gencode/release35_GRCh38.p13/fasta/GRCh38.p13.genome.fa \
/projects/qbio/bifo/users/sergio/reference/gencode/release35_GRCh38.p13/rsem_bowtie2/GRCh38.p13.genome
```


#### Align

```bash
cd /projects/qbio/bifo/users/sergio/repository/20201122_hush/rnaseq/fastq_trimmed

mkdir ../rsem

bwtidx=/projects/qbio/bifo/users/sergio/reference/gencode/release35_GRCh38.p13/rsem_bowtie2/GRCh38.p13.genome

for fq1 in *_1.fastq.gz
do
  fq2=${fq1/_1./_2.}
  bname=${fq1%_1.fastq.gz}
  #echo $fq1, $fq2, $bname
  sbatch -J $bname -o ../rsem/$bname.log --mem 32G -t 3-0 --wrap "rsem-calculate-expression \
  --paired-end \
  -p 16 \
  --bowtie2 \
  --append-names \
  --output-genome-bam \
  --sort-bam-by-coordinate \
  $fq1 \
  $fq2 \
  $bwtidx \
  ../rsem/$bname"
done

cd ../rsem

tail *.log

grep "overall alignment rate" *.log
# MORC2-KO_rep1_SRR5287126.log:44.62% overall alignment rate
# MORC2-KO_rep2_SRR5287127.log:40.10% overall alignment rate
# MPP8-KO_rep1_SRR5287128.log:43.14% overall alignment rate
# MPP8-KO_rep2_SRR5287129.log:45.78% overall alignment rate
# TASOR-KO_rep1_SRR5287130.log:47.10% overall alignment rate
# TASOR-KO_rep2_SRR5287131.log:47.48% overall alignment rate
# WT_rep1_SRR5287124.log:42.37% overall alignment rate
# WT_rep2_SRR5287125.log:39.16% overall alignment rate

wc -l *.genes.results
   # 60657 MORC2-KO_rep1_SRR5287126.genes.results
   # 60657 MORC2-KO_rep2_SRR5287127.genes.results
   # 60657 MPP8-KO_rep1_SRR5287128.genes.results
   # 60657 MPP8-KO_rep2_SRR5287129.genes.results
   # 60657 TASOR-KO_rep1_SRR5287130.genes.results
   # 60657 TASOR-KO_rep2_SRR5287131.genes.results
   # 60657 WT_rep1_SRR5287124.genes.results
   # 60657 WT_rep2_SRR5287125.genes.results
```



### Alignment (hisat2)

From Liu2018: RNA-seq reads were aligned to the hg38 reference genome with hisat2 (–no-mixed,–no-discordant) without constraining to known transcriptome. Known (gencode 25) and de novo transcript coverages were quantified with featureCount.

#### Prepare index

```bash
module load bcbio-nextgen/1.1.0
cd /projects/qbio/bifo/users/sergio/reference/gencode/release35_GRCh38.p13/fasta
sbatch -J hisat2-build --mem 8G -t 1-0 --wrap "hisat2-build GRCh38.p13.genome.fa GRCh38.p13.genome"
rm slurm-61863326.out
```

#### Align, sort and index

```bash
cd /projects/qbio/bifo/users/sergio/repository/20201122_hush/rnaseq/fastq_trimmed

mkdir ../hisat2

idx=/projects/qbio/bifo/users/sergio/reference/gencode/release35_GRCh38.p13/fasta/GRCh38.p13.genome

for fq1 in *_1.fastq.gz
do
  fq2=${fq1/_1./_2.}
  bname=${fq1%_1.fastq.gz}
  #echo $fq1, $fq2, $bname
  sbatch -J $bname -o ../hisat2/$bname.log --mem 32G -t 3-0 --wrap "hisat2 --no-mixed --no-discordant -p 8 -x $idx -1 $fq1 -2 $fq2 | \
  samtools view -@ 8 -bS - | \
  samtools sort -@ 8 -T /projects/qbio/bifo/users/sergio/tmp/$bname -o ../hisat2/$bname.bam - && \
  samtools index ../hisat2/$bname.bam"
done

cd ../hisat2

tail *.log

grep "overall alignment rate" *.log
# MORC2-KO_rep1_SRR5287126.log:86.88% overall alignment rate
# MORC2-KO_rep2_SRR5287127.log:83.62% overall alignment rate
# MPP8-KO_rep1_SRR5287128.log:85.95% overall alignment rate
# MPP8-KO_rep2_SRR5287129.log:86.40% overall alignment rate
# TASOR-KO_rep1_SRR5287130.log:85.57% overall alignment rate
# TASOR-KO_rep2_SRR5287131.log:86.67% overall alignment rate
# WT_rep1_SRR5287124.log:86.01% overall alignment rate
# WT_rep2_SRR5287125.log:87.39% overall alignment rate
```



### bigwig

```bash
cd /projects/qbio/bifo/users/sergio/repository/20201122_hush/rnaseq/hisat2

mkdir ../bigwig

for bam in *.bam
do
  bname=${bam%.bam}
  sbatch -J $bname -o ../bigwig/$bname.log --mem 64G -t 1-0 --wrap "bamCoverage -b $bam -o ../bigwig/${bname}.bw -of bigwig --binSize 10 --normalizeUsing RPKM"
done

tail ../bigwig/*.log
```



### htseq-count

In order for `htseq-count` to work, I need to remove `/projects/qbio/bifo/python3/lib/python3.6/site-packages` from PYTHONPATH so that `numpy` does not get confused.
The same applies to `deeptools`

```bash
cd /projects/qbio/bifo/users/sergio/repository/20201122_hush/rnaseq/hisat2

mkdir ../htseq

gtf='/projects/qbio/bifo/users/sergio/reference/gencode/release35_GRCh38.p13/gtf/gencode.v35.annotation.gtf'

for bam in *.bam
do
	bname=${bam%.bam}
	sbatch -J $bname -o ../htseq/$bname.log --mem 128G -t 1-0 --wrap "htseq-count -f bam -r pos -s no -t exon -i gene_id -m intersection-strict $bam $gtf > ../htseq/$bname.txt"
done

cd ../htseq
tail *.log # looks fine

ls *.txt | xargs wc -l
```



### bedtools multicov

```bash
cd /projects/qbio/bifo/users/sergio/repository/20201122_hush/rnaseq/hisat2

mkdir ../multicov

bed=/projects/qbio/bifo/users/sergio/repository/20201122_hush/annotation/gencode.v35.annotation.gene_type.intronless.bed

for bam in *.bam
do
	bname=${bam%.bam}
	sbatch -J $bname -o ../multicov/$bname.log --mem 32G -t 1-0 --wrap "bedtools multicov -bams $bam -bed $bed -D | cut -f5,9 > ../multicov/$bname.txt"
done

cd ../multicov
tail *.log # looks fine

ls *.txt | xargs wc -l
   # 60656 MORC2-KO_rep1_SRR5287126.txt
   # 60656 MORC2-KO_rep2_SRR5287127.txt
   # 60656 MPP8-KO_rep1_SRR5287128.txt
   # 60656 MPP8-KO_rep2_SRR5287129.txt
   # 60656 TASOR-KO_rep1_SRR5287130.txt
   # 60656 TASOR-KO_rep2_SRR5287131.txt
   # 60656 WT_rep1_SRR5287124.txt
   # 60656 WT_rep2_SRR5287125.txt


grep "ENSG00000253797.2" *.txt
# MORC2-KO_rep1_SRR5287126.txt:ENSG00000253797.2	3508
# MORC2-KO_rep2_SRR5287127.txt:ENSG00000253797.2	2207
# MPP8-KO_rep1_SRR5287128.txt:ENSG00000253797.2	3748
# MPP8-KO_rep2_SRR5287129.txt:ENSG00000253797.2	2424
# TASOR-KO_rep1_SRR5287130.txt:ENSG00000253797.2	3747
# TASOR-KO_rep2_SRR5287131.txt:ENSG00000253797.2	3315
# WT_rep1_SRR5287124.txt:ENSG00000253797.2	3544
# WT_rep2_SRR5287125.txt:ENSG00000253797.2	1760
```





## ChIP-seq

K562 datasets:

GSE | GSM name | SRR
----|----------|-----
GSE95374 | GSM2509455	ChIP:MORC2_Cell:WT_rep1 | SRR5287068
GSE95374 | GSM2509456	ChIP:MORC2_Cell:WT_rep2 | SRR5287069
GSE95374 | GSM2509457	ChIP:MORC2_Cell:MORC2-KO_rep1 | SRR5287070
GSE95374 | GSM2509458	ChIP:MORC2_Cell:MORC2-KO_rep2 | SRR5287071
GSE95374 | GSM2509459	ChIP:MORC2_Cell:MPP8-KO_rep1 | SRR5287072
GSE95374 | GSM2509460	ChIP:MORC2_Cell:MPP8-KO_rep2 | SRR5287073
GSE95374 | GSM2509461	ChIP:MORC2_Cell:TASOR-KO_rep1 | SRR5287074
GSE95374 | GSM2509462	ChIP:MORC2_Cell:TASOR-KO_rep2 | SRR5287075
GSE95374 | GSM2509463	ChIP:MPP8_Cell:WT_rep1 | SRR5287076
GSE95374 | GSM2509464	ChIP:MPP8_Cell:WT_rep2 | SRR5287077
GSE95374 | GSM2509465	ChIP:MPP8_Cell:MORC2-KO_rep1 | SRR5287078
GSE95374 | GSM2509466	ChIP:MPP8_Cell:MORC2-KO_rep2 | SRR5287079
GSE95374 | GSM2509467	ChIP:MPP8_Cell:MPP8-KO_rep1 | SRR5287080
GSE95374 | GSM2509468	ChIP:MPP8_Cell:MPP8-KO_rep2 | SRR5287081
GSE95374 | GSM2509469	ChIP:MPP8_Cell:TASOR-KO_rep1 | SRR5287082
GSE95374 | GSM2509470	ChIP:MPP8_Cell:TASOR-KO_rep2 | SRR5287083
GSE95374 | GSM2509471	ChIP:TASOR_Cell:WT_rep1 | SRR5287084
GSE95374 | GSM2509472	ChIP:TASOR_Cell:WT_rep2 | SRR5287085
GSE95374 | GSM2509473	ChIP:TASOR_Cell:MORC2-KO_rep1 | SRR5287086
GSE95374 | GSM2509474	ChIP:TASOR_Cell:MORC2-KO_rep2 | SRR5287087
GSE95374 | GSM2509475	ChIP:TASOR_Cell:MPP8-KO_rep1 | SRR5287088
GSE95374 | GSM2509476	ChIP:TASOR_Cell:MPP8-KO_rep2 | SRR5287089
GSE95374 | GSM2509477	ChIP:TASOR_Cell:TASOR-KO_rep1 | SRR5287090
GSE95374 | GSM2509478	ChIP:TASOR_Cell:TASOR-KO_rep2 | SRR5287091
GSE95374 | GSM2509479	ChIP:Input(MORC2, MPP8, TASOR)_Cell:WT_rep1 | SRR5287092
GSE95374 | GSM2509480	ChIP:Input(MORC2, MPP8, TASOR)_Cell:WT_rep2 | SRR5287093
GSE95374 | GSM2509481	ChIP:Input(MORC2, MPP8, TASOR)_Cell:MORC2-KO_rep1 | SRR5287094
GSE95374 | GSM2509482	ChIP:Input(MORC2, MPP8, TASOR)_Cell:MORC2-KO_rep2 | SRR5287095
GSE95374 | GSM2509483	ChIP:Input(MORC2, MPP8, TASOR)_Cell:MPP8-KO_rep1 | SRR5287096
GSE95374 | GSM2509484	ChIP:Input(MORC2, MPP8, TASOR)_Cell:MPP8-KO_rep2 | SRR5287097
GSE95374 | GSM2509485	ChIP:Input(MORC2, MPP8, TASOR)_Cell:TASOR-KO_rep1 | SRR5287098
GSE95374 | GSM2509486	ChIP:Input(MORC2, MPP8, TASOR)_Cell:TASOR-KO_rep2 | SRR5287099
GSE95374 | GSM2509487	ChIP:H3K9me3_Cell:WT_rep1 | SRR5287100
GSE95374 | GSM2509488	ChIP:H3K9me3_Cell:WT_rep2 | SRR5287101
GSE95374 | GSM2509489	ChIP:H3K9me3_Cell:MORC2-KO_rep1 | SRR5287102
GSE95374 | GSM2509490	ChIP:H3K9me3_Cell:MORC2-KO_rep2 | SRR5287103
GSE95374 | GSM2509491	ChIP:H3K9me3_Cell:TASOR-KO_rep1 | SRR5287104
GSE95374 | GSM2509492	ChIP:H3K9me3_Cell:TASOR-KO_rep2 | SRR5287105
GSE95374 | GSM2509493	ChIP:Input(H3K9me3)_Cell:WT_rep1 | SRR5287106
GSE95374 | GSM2509494	ChIP:Input(H3K9me3)_Cell:WT_rep2 | SRR5287107
GSE95374 | GSM2509495	ChIP:Input(H3K9me3)_Cell:MORC2-KO_rep1 | SRR5287108
GSE95374 | GSM2509496	ChIP:Input(H3K9me3)_Cell:MORC2-KO_rep2 | SRR5287109
GSE95374 | GSM2509497	ChIP:Input(H3K9me3)_Cell:TASOR-KO_rep1 | SRR5287110
GSE95374 | GSM2509498	ChIP:Input(H3K9me3)_Cell:TASOR-KO_rep2 | SRR5287111
GSE95374 | GSM2509503	ChIP:H3K9me3_Cell:WT_rep3 | SRR5287116
GSE95374 | GSM2509504	ChIP:H3K9me3_Cell:WT_rep4 | SRR5287117
GSE95374 | GSM2509505	ChIP:H3K9me3_Cell:MPP8-KO_rep1 | SRR5287118
GSE95374 | GSM2509506	ChIP:H3K9me3_Cell:MPP8-KO_rep2 | SRR5287119
GSE95374 | GSM2509507	ChIP:Input(H3K9me2or3)_Cell:WT_rep1 | SRR5287120
GSE95374 | GSM2509508	ChIP:Input(H3K9me2or3)_Cell:WT_rep2 | SRR5287121
GSE95374 | GSM2509509	ChIP:Input(H3K9me2or3)_Cell:MPP8-KO_rep1 | SRR5287122
GSE95374 | GSM2509510	ChIP:Input(H3K9me2or3)_Cell:MPP8-KO_rep2 | SRR5287123



### Download fastq files

```bash
cd /projects/qbio/bifo/users/sergio/repository/20201122_hush

mkdir -p chipseq/fastq && cd chipseq/fastq

for run_id in SRR52870{68..99} SRR52871{00..11} SRR52871{16..23}
do
  sbatch -J $run_id -o $run_id.log --mem 32G -t 1-0 --wrap "fastq-dump --outdir . --gzip --split-files -F $run_id"
done

tail *.log
```



### Rename fastq files

```bash
cd /projects/qbio/bifo/users/sergio/repository/20201122_hush/chipseq/fastq

declare -A pairs=(\
[SRR5287068]=ChIP_MORC2_WT_rep1 \
[SRR5287069]=ChIP_MORC2_WT_rep2 \
[SRR5287070]=ChIP_MORC2_MORC2-KO_rep1 \
[SRR5287071]=ChIP_MORC2_MORC2-KO_rep2 \
[SRR5287072]=ChIP_MORC2_MPP8-KO_rep1 \
[SRR5287073]=ChIP_MORC2_MPP8-KO_rep2 \
[SRR5287074]=ChIP_MORC2_TASOR-KO_rep1 \
[SRR5287075]=ChIP_MORC2_TASOR-KO_rep2 \
[SRR5287076]=ChIP_MPP8_WT_rep1 \
[SRR5287077]=ChIP_MPP8_WT_rep2 \
[SRR5287078]=ChIP_MPP8_MORC2-KO_rep1 \
[SRR5287079]=ChIP_MPP8_MORC2-KO_rep2 \
[SRR5287080]=ChIP_MPP8_MPP8-KO_rep1 \
[SRR5287081]=ChIP_MPP8_MPP8-KO_rep2 \
[SRR5287082]=ChIP_MPP8_TASOR-KO_rep1 \
[SRR5287083]=ChIP_MPP8_TASOR-KO_rep2 \
[SRR5287084]=ChIP_TASOR_WT_rep1 \
[SRR5287085]=ChIP_TASOR_WT_rep2 \
[SRR5287086]=ChIP_TASOR_MORC2-KO_rep1 \
[SRR5287087]=ChIP_TASOR_MORC2-KO_rep2 \
[SRR5287088]=ChIP_TASOR_MPP8-KO_rep1 \
[SRR5287089]=ChIP_TASOR_MPP8-KO_rep2 \
[SRR5287090]=ChIP_TASOR_TASOR-KO_rep1 \
[SRR5287091]=ChIP_TASOR_TASOR-KO_rep2 \
[SRR5287092]=Input_MORC2_WT_rep1 \
[SRR5287093]=Input_MORC2_WT_rep2 \
[SRR5287094]=Input_MORC2_MORC2-KO_rep1 \
[SRR5287095]=Input_MORC2_MORC2-KO_rep2 \
[SRR5287096]=Input_MORC2_MPP8-KO_rep1 \
[SRR5287097]=Input_MORC2_MPP8-KO_rep2 \
[SRR5287098]=Input_MORC2_TASOR-KO_rep1 \
[SRR5287099]=Input_MORC2_TASOR-KO_rep2 \
[SRR5287100]=ChIP_H3K9me3_WT_rep1 \
[SRR5287101]=ChIP_H3K9me3_WT_rep2 \
[SRR5287102]=ChIP_H3K9me3_MORC2-KO_rep1 \
[SRR5287103]=ChIP_H3K9me3_MORC2-KO_rep2 \
[SRR5287104]=ChIP_H3K9me3_TASOR-KO_rep1 \
[SRR5287105]=ChIP_H3K9me3_TASOR-KO_rep2 \
[SRR5287106]=Input_H3K9me3_WT_rep1 \
[SRR5287107]=Input_H3K9me3_WT_rep2 \
[SRR5287108]=Input_H3K9me3_MORC2-KO_rep1 \
[SRR5287109]=Input_H3K9me3_MORC2-KO_rep2 \
[SRR5287110]=Input_H3K9me3_TASOR-KO_rep1 \
[SRR5287111]=Input_H3K9me3_TASOR-KO_rep2 \
[SRR5287116]=ChIP_H3K9me3_WT_rep3 \
[SRR5287117]=ChIP_H3K9me3_WT_rep4 \
[SRR5287118]=ChIP_H3K9me3_MPP8-KO_rep1 \
[SRR5287119]=ChIP_H3K9me3_MPP8-KO_rep2 \
[SRR5287120]=Input_H3K9me2or3_WT_rep1 \
[SRR5287121]=Input_H3K9me2or3_WT_rep2 \
[SRR5287122]=Input_H3K9me2or3_MPP8-KO_rep1 \
[SRR5287123]=Input_H3K9me2or3_MPP8-KO_rep2)

for id in ${!pairs[@]}
do
  for filename in ${id}_{1,2}.fastq.gz; do mv "$filename" "${pairs[$id]}_$filename"; done;
done
```



### Trim illumina adapters and quality trimming

```bash
cd /projects/qbio/bifo/users/sergio/repository/20201122_hush/chipseq/fastq

mkdir ../fastq_trimmed

for fq1 in *_1.fastq.gz
do
  fq2=${fq1/_1./_2.}
  bname=${fq1%_1.fastq.gz}
  #echo $fq1, $fq2, $bname
  sbatch -J $bname -o ../fastq_trimmed/$bname.log --mem 4G -t 1-0 --wrap "cutadapt -a AGATCGGAAGAGC -A AGATCGGAAGAGC -m 50 -q 10 -o ../fastq_trimmed/$fq1 -p ../fastq_trimmed/$fq2 $fq1 $fq2 > ../fastq_trimmed/$bname.txt"
done

cd ../fastq_trimmed

tail *.log

grep "Read 1 with adapter" *.txt
grep "Read 2 with adapter" *.txt
grep "Pairs written (passing filters):" *.txt
```



### Alignment (using rsem indexes)

#### Align and sort

```bash
cd /projects/qbio/bifo/users/sergio/repository/20201122_hush/chipseq/fastq_trimmed

mkdir ../bam

ref=/projects/qbio/bifo/users/sergio/reference/gencode/release35_GRCh38.p13/rsem_bowtie2/GRCh38.p13.genome

for fq1 in *_1.fastq.gz
do
  bname=${fq1%_1.fastq.gz}
  fq2=${fq1/_1./_2.}
  #echo $fq1,$fq2
  sbatch -J $bname -o ../bam/$bname.log --mem 32G -t 1-0 --wrap "bowtie2 --no-mixed --no-discordant --end-to-end --maxins 500 -x $ref -1 $fq1 -2 $fq2 | \
  samtools view -@ 20 -bS - | \
  samtools sort -@ 20 -T /projects/qbio/bifo/users/sergio/tmp/$bname -o ../bam/$bname.bam -"
done

cd ../bam
grep "overall alignment rate" *.log
# ChIP_H3K9me3_MORC2-KO_rep1_SRR5287102.log:12.46% overall alignment rate
# ChIP_H3K9me3_MORC2-KO_rep2_SRR5287103.log:12.93% overall alignment rate
# ChIP_H3K9me3_MPP8-KO_rep1_SRR5287118.log:11.40% overall alignment rate
# ChIP_H3K9me3_MPP8-KO_rep2_SRR5287119.log:11.72% overall alignment rate
# ChIP_H3K9me3_TASOR-KO_rep1_SRR5287104.log:12.68% overall alignment rate
# ChIP_H3K9me3_TASOR-KO_rep2_SRR5287105.log:12.94% overall alignment rate
# ChIP_H3K9me3_WT_rep1_SRR5287100.log:11.68% overall alignment rate
# ChIP_H3K9me3_WT_rep2_SRR5287101.log:12.26% overall alignment rate
# ChIP_H3K9me3_WT_rep3_SRR5287116.log:12.10% overall alignment rate
# ChIP_H3K9me3_WT_rep4_SRR5287117.log:11.95% overall alignment rate
# ChIP_MORC2_MORC2-KO_rep1_SRR5287070.log:8.86% overall alignment rate
# ChIP_MORC2_MORC2-KO_rep2_SRR5287071.log:8.70% overall alignment rate
# ChIP_MORC2_MPP8-KO_rep1_SRR5287072.log:8.98% overall alignment rate
# ChIP_MORC2_MPP8-KO_rep2_SRR5287073.log:8.84% overall alignment rate
# ChIP_MORC2_TASOR-KO_rep1_SRR5287074.log:8.70% overall alignment rate
# ChIP_MORC2_TASOR-KO_rep2_SRR5287075.log:8.69% overall alignment rate
# ChIP_MORC2_WT_rep1_SRR5287068.log:10.54% overall alignment rate
# ChIP_MORC2_WT_rep2_SRR5287069.log:10.55% overall alignment rate
# ChIP_MPP8_MORC2-KO_rep1_SRR5287078.log:8.79% overall alignment rate
# ChIP_MPP8_MORC2-KO_rep2_SRR5287079.log:8.58% overall alignment rate
# ChIP_MPP8_MPP8-KO_rep1_SRR5287080.log:8.80% overall alignment rate
# ChIP_MPP8_MPP8-KO_rep2_SRR5287081.log:8.04% overall alignment rate
# ChIP_MPP8_TASOR-KO_rep1_SRR5287082.log:8.45% overall alignment rate
# ChIP_MPP8_TASOR-KO_rep2_SRR5287083.log:8.46% overall alignment rate
# ChIP_MPP8_WT_rep1_SRR5287076.log:8.51% overall alignment rate
# ChIP_MPP8_WT_rep2_SRR5287077.log:8.84% overall alignment rate
# ChIP_TASOR_MORC2-KO_rep1_SRR5287086.log:8.90% overall alignment rate
# ChIP_TASOR_MORC2-KO_rep2_SRR5287087.log:8.70% overall alignment rate
# ChIP_TASOR_MPP8-KO_rep1_SRR5287088.log:8.98% overall alignment rate
# ChIP_TASOR_MPP8-KO_rep2_SRR5287089.log:8.39% overall alignment rate
# ChIP_TASOR_TASOR-KO_rep1_SRR5287090.log:8.42% overall alignment rate
# ChIP_TASOR_TASOR-KO_rep2_SRR5287091.log:8.59% overall alignment rate
# ChIP_TASOR_WT_rep1_SRR5287084.log:8.48% overall alignment rate
# ChIP_TASOR_WT_rep2_SRR5287085.log:8.64% overall alignment rate
# Input_H3K9me2or3_MPP8-KO_rep1_SRR5287122.log:10.10% overall alignment rate
# Input_H3K9me2or3_MPP8-KO_rep2_SRR5287123.log:10.47% overall alignment rate
# Input_H3K9me2or3_WT_rep1_SRR5287120.log:9.48% overall alignment rate
# Input_H3K9me2or3_WT_rep2_SRR5287121.log:9.84% overall alignment rate
# Input_H3K9me3_MORC2-KO_rep1_SRR5287108.log:16.12% overall alignment rate
# Input_H3K9me3_MORC2-KO_rep2_SRR5287109.log:17.13% overall alignment rate
# Input_H3K9me3_TASOR-KO_rep1_SRR5287110.log:13.71% overall alignment rate
# Input_H3K9me3_TASOR-KO_rep2_SRR5287111.log:12.48% overall alignment rate
# Input_H3K9me3_WT_rep1_SRR5287106.log:12.63% overall alignment rate
# Input_H3K9me3_WT_rep2_SRR5287107.log:12.49% overall alignment rate
# Input_MORC2_MORC2-KO_rep1_SRR5287094.log:11.17% overall alignment rate
# Input_MORC2_MORC2-KO_rep2_SRR5287095.log:11.12% overall alignment rate
# Input_MORC2_MPP8-KO_rep1_SRR5287096.log:12.76% overall alignment rate
# Input_MORC2_MPP8-KO_rep2_SRR5287097.log:9.89% overall alignment rate
# Input_MORC2_TASOR-KO_rep1_SRR5287098.log:9.85% overall alignment rate
# Input_MORC2_TASOR-KO_rep2_SRR5287099.log:10.63% overall alignment rate
# Input_MORC2_WT_rep1_SRR5287092.log:8.69% overall alignment rate
# Input_MORC2_WT_rep2_SRR5287093.log:9.74% overall alignment rate
```



### Alignment (using bowtie2 indexes)

#### Prepare reference

```bash
cd /projects/qbio/bifo/users/sergio/reference/gencode/release35_GRCh38.p13/fasta
module load bcbio-nextgen/1.1.0
sbatch -J bowtie2-build --mem 8G -t 1-0 --wrap "bowtie2-build GRCh38.p13.genome.fa GRCh38.p13.genome"
rm slurm-61695468.out
```

#### Align and sort

```bash
cd /projects/qbio/bifo/users/sergio/repository/20201122_hush/chipseq/fastq_trimmed

mkdir ../bam_bowtie2

ref=/projects/qbio/bifo/users/sergio/reference/gencode/release35_GRCh38.p13/fasta/GRCh38.p13.genome

for fq1 in *_1.fastq.gz
do
  bname=${fq1%_1.fastq.gz}
  fq2=${fq1/_1./_2.}
  #echo $fq1,$fq2
  sbatch -J $bname -o ../bam_bowtie2/$bname.log --mem 32G -t 1-0 --wrap "bowtie2 --no-mixed --no-discordant --end-to-end --maxins 500 -x $ref -1 $fq1 -2 $fq2 | \
  samtools view -@ 20 -bS - | \
  samtools sort -@ 20 -T /projects/qbio/bifo/users/sergio/tmp/$bname -o ../bam_bowtie2/$bname.bam -"
done

cd ../bam_bowtie2
grep "overall alignment rate" *.log
# ChIP_H3K9me3_MORC2-KO_rep1_SRR5287102.log:94.71% overall alignment rate
# ChIP_H3K9me3_MORC2-KO_rep2_SRR5287103.log:94.83% overall alignment rate
# ChIP_H3K9me3_MPP8-KO_rep1_SRR5287118.log:96.26% overall alignment rate
# ChIP_H3K9me3_MPP8-KO_rep2_SRR5287119.log:95.44% overall alignment rate
# ChIP_H3K9me3_TASOR-KO_rep1_SRR5287104.log:94.71% overall alignment rate
# ChIP_H3K9me3_TASOR-KO_rep2_SRR5287105.log:94.49% overall alignment rate
# ChIP_H3K9me3_WT_rep1_SRR5287100.log:95.34% overall alignment rate
# ChIP_H3K9me3_WT_rep2_SRR5287101.log:94.99% overall alignment rate
# ChIP_H3K9me3_WT_rep3_SRR5287116.log:96.39% overall alignment rate
# ChIP_H3K9me3_WT_rep4_SRR5287117.log:96.64% overall alignment rate
# ChIP_MORC2_MORC2-KO_rep1_SRR5287070.log:96.35% overall alignment rate
# ChIP_MORC2_MORC2-KO_rep2_SRR5287071.log:96.91% overall alignment rate
# ChIP_MORC2_MPP8-KO_rep1_SRR5287072.log:96.93% overall alignment rate
# ChIP_MORC2_MPP8-KO_rep2_SRR5287073.log:96.90% overall alignment rate
# ChIP_MORC2_TASOR-KO_rep1_SRR5287074.log:96.68% overall alignment rate
# ChIP_MORC2_TASOR-KO_rep2_SRR5287075.log:96.79% overall alignment rate
# ChIP_MORC2_WT_rep1_SRR5287068.log:97.22% overall alignment rate
# ChIP_MORC2_WT_rep2_SRR5287069.log:96.46% overall alignment rate
# ChIP_MPP8_MORC2-KO_rep1_SRR5287078.log:96.68% overall alignment rate
# ChIP_MPP8_MORC2-KO_rep2_SRR5287079.log:96.74% overall alignment rate
# ChIP_MPP8_MPP8-KO_rep1_SRR5287080.log:96.85% overall alignment rate
# ChIP_MPP8_MPP8-KO_rep2_SRR5287081.log:96.49% overall alignment rate
# ChIP_MPP8_TASOR-KO_rep1_SRR5287082.log:96.81% overall alignment rate
# ChIP_MPP8_TASOR-KO_rep2_SRR5287083.log:96.76% overall alignment rate
# ChIP_MPP8_WT_rep1_SRR5287076.log:96.34% overall alignment rate
# ChIP_MPP8_WT_rep2_SRR5287077.log:96.80% overall alignment rate
# ChIP_TASOR_MORC2-KO_rep1_SRR5287086.log:97.13% overall alignment rate
# ChIP_TASOR_MORC2-KO_rep2_SRR5287087.log:96.92% overall alignment rate
# ChIP_TASOR_MPP8-KO_rep1_SRR5287088.log:96.71% overall alignment rate
# ChIP_TASOR_MPP8-KO_rep2_SRR5287089.log:96.49% overall alignment rate
# ChIP_TASOR_TASOR-KO_rep1_SRR5287090.log:96.43% overall alignment rate
# ChIP_TASOR_TASOR-KO_rep2_SRR5287091.log:96.34% overall alignment rate
# ChIP_TASOR_WT_rep1_SRR5287084.log:96.71% overall alignment rate
# ChIP_TASOR_WT_rep2_SRR5287085.log:96.85% overall alignment rate
# Input_H3K9me2or3_MPP8-KO_rep1_SRR5287122.log:97.08% overall alignment rate
# Input_H3K9me2or3_MPP8-KO_rep2_SRR5287123.log:95.04% overall alignment rate
# Input_H3K9me2or3_WT_rep1_SRR5287120.log:96.77% overall alignment rate
# Input_H3K9me2or3_WT_rep2_SRR5287121.log:96.38% overall alignment rate
# Input_H3K9me3_MORC2-KO_rep1_SRR5287108.log:74.44% overall alignment rate
# Input_H3K9me3_MORC2-KO_rep2_SRR5287109.log:89.61% overall alignment rate
# Input_H3K9me3_TASOR-KO_rep1_SRR5287110.log:95.93% overall alignment rate
# Input_H3K9me3_TASOR-KO_rep2_SRR5287111.log:96.47% overall alignment rate
# Input_H3K9me3_WT_rep1_SRR5287106.log:96.23% overall alignment rate
# Input_H3K9me3_WT_rep2_SRR5287107.log:96.60% overall alignment rate
# Input_MORC2_MORC2-KO_rep1_SRR5287094.log:95.57% overall alignment rate
# Input_MORC2_MORC2-KO_rep2_SRR5287095.log:95.28% overall alignment rate
# Input_MORC2_MPP8-KO_rep1_SRR5287096.log:91.20% overall alignment rate
# Input_MORC2_MPP8-KO_rep2_SRR5287097.log:96.76% overall alignment rate
# Input_MORC2_TASOR-KO_rep1_SRR5287098.log:96.92% overall alignment rate
# Input_MORC2_TASOR-KO_rep2_SRR5287099.log:96.49% overall alignment rate
# Input_MORC2_WT_rep1_SRR5287092.log:96.71% overall alignment rate
# Input_MORC2_WT_rep2_SRR5287093.log:95.91% overall alignment rate
```

#### Index

```bash
cd /projects/qbio/bifo/users/sergio/repository/20201122_hush/chipseq/bam_bowtie2

for bam in *.bam
do
  bname=${bam%.bam}
  sbatch -J $bname -o $bname.index.log --mem 8G -t 1-0 --wrap "samtools index $bam"
done

tail *.index.log
```



### bigwig

```bash
module load bcbio-nextgen/1.1.0

cd /projects/qbio/bifo/users/sergio/repository/20201122_hush/chipseq/bam_bowtie2

mkdir ../bigwig

for bam in *.bam
do
  bname=${bam%.bam}
  sbatch -J $bname -o ../bigwig/$bname.log --mem 8G -t 1-0 --wrap "bamCoverage -b $bam -o ../bigwig/${bname}.bw -of bigwig --binSize 10 --normalizeUsing RPKM"
done

tail ../bigwig/*.log
```



### htseq-count

In order for `htseq-count` to work, I need to remove `/projects/qbio/bifo/python3/lib/python3.6/site-packages` from PYTHONPATH so that `numpy` does not get confused.
The same applies to `deeptools`

```bash
cd /projects/qbio/bifo/users/sergio/repository/20201122_hush/chipseq/bam_bowtie2

mkdir ../htseq

gtf='/projects/qbio/bifo/users/sergio/reference/gencode/release35_GRCh38.p13/gtf/gencode.v35.annotation.gtf'

for bam in *.bam
do
	bname=${bam%.bam}
	sbatch -J $bname -o ../htseq/$bname.log --mem 16G -t 1-0 --wrap "htseq-count -f bam -r pos -s no -t exon -i gene_id -m intersection-strict $bam $gtf > ../htseq/$bname.txt"
done

cd ../htseq
tail *.log # looks fine

ls *.txt | xargs wc -l
   # 60661 ChIP_H3K9me3_MORC2-KO_rep1_SRR5287102.txt
   # 60661 ChIP_H3K9me3_MORC2-KO_rep2_SRR5287103.txt
   # 60661 ChIP_H3K9me3_MPP8-KO_rep1_SRR5287118.txt
   # 60661 ChIP_H3K9me3_MPP8-KO_rep2_SRR5287119.txt
   # 60661 ChIP_H3K9me3_TASOR-KO_rep1_SRR5287104.txt
   # 60661 ChIP_H3K9me3_TASOR-KO_rep2_SRR5287105.txt
   # 60661 ChIP_H3K9me3_WT_rep1_SRR5287100.txt
   # 60661 ChIP_H3K9me3_WT_rep2_SRR5287101.txt
   # 60661 ChIP_H3K9me3_WT_rep3_SRR5287116.txt
   # 60661 ChIP_H3K9me3_WT_rep4_SRR5287117.txt
   # 60661 ChIP_MORC2_MORC2-KO_rep1_SRR5287070.txt
   # 60661 ChIP_MORC2_MORC2-KO_rep2_SRR5287071.txt
   # 60661 ChIP_MORC2_MPP8-KO_rep1_SRR5287072.txt
   # 60661 ChIP_MORC2_MPP8-KO_rep2_SRR5287073.txt
   # 60661 ChIP_MORC2_TASOR-KO_rep1_SRR5287074.txt
   # 60661 ChIP_MORC2_TASOR-KO_rep2_SRR5287075.txt
   # 60661 ChIP_MORC2_WT_rep1_SRR5287068.txt
   # 60661 ChIP_MORC2_WT_rep2_SRR5287069.txt
   # 60661 ChIP_MPP8_MORC2-KO_rep1_SRR5287078.txt
   # 60661 ChIP_MPP8_MORC2-KO_rep2_SRR5287079.txt
   # 60661 ChIP_MPP8_MPP8-KO_rep1_SRR5287080.txt
   # 60661 ChIP_MPP8_MPP8-KO_rep2_SRR5287081.txt
   # 60661 ChIP_MPP8_TASOR-KO_rep1_SRR5287082.txt
   # 60661 ChIP_MPP8_TASOR-KO_rep2_SRR5287083.txt
   # 60661 ChIP_MPP8_WT_rep1_SRR5287076.txt
   # 60661 ChIP_MPP8_WT_rep2_SRR5287077.txt
   # 60661 ChIP_TASOR_MORC2-KO_rep1_SRR5287086.txt
   # 60661 ChIP_TASOR_MORC2-KO_rep2_SRR5287087.txt
   # 60661 ChIP_TASOR_MPP8-KO_rep1_SRR5287088.txt
   # 60661 ChIP_TASOR_MPP8-KO_rep2_SRR5287089.txt
   # 60661 ChIP_TASOR_TASOR-KO_rep1_SRR5287090.txt
   # 60661 ChIP_TASOR_TASOR-KO_rep2_SRR5287091.txt
   # 60661 ChIP_TASOR_WT_rep1_SRR5287084.txt
   # 60661 ChIP_TASOR_WT_rep2_SRR5287085.txt
   # 60661 Input_H3K9me2or3_MPP8-KO_rep1_SRR5287122.txt
   # 60661 Input_H3K9me2or3_MPP8-KO_rep2_SRR5287123.txt
   # 60661 Input_H3K9me2or3_WT_rep1_SRR5287120.txt
   # 60661 Input_H3K9me2or3_WT_rep2_SRR5287121.txt
   # 60661 Input_H3K9me3_MORC2-KO_rep1_SRR5287108.txt
   # 60661 Input_H3K9me3_MORC2-KO_rep2_SRR5287109.txt
   # 60661 Input_H3K9me3_TASOR-KO_rep1_SRR5287110.txt
   # 60661 Input_H3K9me3_TASOR-KO_rep2_SRR5287111.txt
   # 60661 Input_H3K9me3_WT_rep1_SRR5287106.txt
   # 60661 Input_H3K9me3_WT_rep2_SRR5287107.txt
   # 60661 Input_MORC2_MORC2-KO_rep1_SRR5287094.txt
   # 60661 Input_MORC2_MORC2-KO_rep2_SRR5287095.txt
   # 60661 Input_MORC2_MPP8-KO_rep1_SRR5287096.txt
   # 60661 Input_MORC2_MPP8-KO_rep2_SRR5287097.txt
   # 60661 Input_MORC2_TASOR-KO_rep1_SRR5287098.txt
   # 60661 Input_MORC2_TASOR-KO_rep2_SRR5287099.txt
   # 60661 Input_MORC2_WT_rep1_SRR5287092.txt
   # 60661 Input_MORC2_WT_rep2_SRR5287093.txt


grep "ENSG00000180660.8" ChIP_H3K9me3_*.txt
# ChIP_H3K9me3_MORC2-KO_rep1_SRR5287102.txt:ENSG00000180660.8	15
# ChIP_H3K9me3_MORC2-KO_rep2_SRR5287103.txt:ENSG00000180660.8	23
# ChIP_H3K9me3_MPP8-KO_rep1_SRR5287118.txt:ENSG00000180660.8	25
# ChIP_H3K9me3_MPP8-KO_rep2_SRR5287119.txt:ENSG00000180660.8	28
# ChIP_H3K9me3_TASOR-KO_rep1_SRR5287104.txt:ENSG00000180660.8	26
# ChIP_H3K9me3_TASOR-KO_rep2_SRR5287105.txt:ENSG00000180660.8	28
# ChIP_H3K9me3_WT_rep1_SRR5287100.txt:ENSG00000180660.8	106
# ChIP_H3K9me3_WT_rep2_SRR5287101.txt:ENSG00000180660.8	133
# ChIP_H3K9me3_WT_rep3_SRR5287116.txt:ENSG00000180660.8	120
# ChIP_H3K9me3_WT_rep4_SRR5287117.txt:ENSG00000180660.8	92
```



### bedtools multicov

```bash
cd /projects/qbio/bifo/users/sergio/repository/20201122_hush/chipseq/bam_bowtie2

mkdir ../multicov

bed=/projects/qbio/bifo/users/sergio/repository/20201122_hush/annotation/gencode.v35.annotation.gene_type.intronless.bed

for bam in *.bam
do
	bname=${bam%.bam}
	sbatch -J $bname -o ../multicov/$bname.log --mem 16G -t 1-0 --wrap "bedtools multicov -bams $bam -bed $bed -D | cut -f5,9 > ../multicov/$bname.txt"
done

cd ../multicov
tail *.log # looks fine

ls *.txt | xargs wc -l
   # 60656 ChIP_H3K9me3_MORC2-KO_rep1_SRR5287102.txt
   # 60656 ChIP_H3K9me3_MORC2-KO_rep2_SRR5287103.txt
   # 60656 ChIP_H3K9me3_MPP8-KO_rep1_SRR5287118.txt
   # 60656 ChIP_H3K9me3_MPP8-KO_rep2_SRR5287119.txt
   # 60656 ChIP_H3K9me3_TASOR-KO_rep1_SRR5287104.txt
   # 60656 ChIP_H3K9me3_TASOR-KO_rep2_SRR5287105.txt
   # 60656 ChIP_H3K9me3_WT_rep1_SRR5287100.txt
   # 60656 ChIP_H3K9me3_WT_rep2_SRR5287101.txt
   # 60656 ChIP_H3K9me3_WT_rep3_SRR5287116.txt
   # 60656 ChIP_H3K9me3_WT_rep4_SRR5287117.txt
   # 60656 ChIP_MORC2_MORC2-KO_rep1_SRR5287070.txt
   # 60656 ChIP_MORC2_MORC2-KO_rep2_SRR5287071.txt
   # 60656 ChIP_MORC2_MPP8-KO_rep1_SRR5287072.txt
   # 60656 ChIP_MORC2_MPP8-KO_rep2_SRR5287073.txt
   # 60656 ChIP_MORC2_TASOR-KO_rep1_SRR5287074.txt
   # 60656 ChIP_MORC2_TASOR-KO_rep2_SRR5287075.txt
   # 60656 ChIP_MORC2_WT_rep1_SRR5287068.txt
   # 60656 ChIP_MORC2_WT_rep2_SRR5287069.txt
   # 60656 ChIP_MPP8_MORC2-KO_rep1_SRR5287078.txt
   # 60656 ChIP_MPP8_MORC2-KO_rep2_SRR5287079.txt
   # 60656 ChIP_MPP8_MPP8-KO_rep1_SRR5287080.txt
   # 60656 ChIP_MPP8_MPP8-KO_rep2_SRR5287081.txt
   # 60656 ChIP_MPP8_TASOR-KO_rep1_SRR5287082.txt
   # 60656 ChIP_MPP8_TASOR-KO_rep2_SRR5287083.txt
   # 60656 ChIP_MPP8_WT_rep1_SRR5287076.txt
   # 60656 ChIP_MPP8_WT_rep2_SRR5287077.txt
   # 60656 ChIP_TASOR_MORC2-KO_rep1_SRR5287086.txt
   # 60656 ChIP_TASOR_MORC2-KO_rep2_SRR5287087.txt
   # 60656 ChIP_TASOR_MPP8-KO_rep1_SRR5287088.txt
   # 60656 ChIP_TASOR_MPP8-KO_rep2_SRR5287089.txt
   # 60656 ChIP_TASOR_TASOR-KO_rep1_SRR5287090.txt
   # 60656 ChIP_TASOR_TASOR-KO_rep2_SRR5287091.txt
   # 60656 ChIP_TASOR_WT_rep1_SRR5287084.txt
   # 60656 ChIP_TASOR_WT_rep2_SRR5287085.txt
   # 60656 Input_H3K9me2or3_MPP8-KO_rep1_SRR5287122.txt
   # 60656 Input_H3K9me2or3_MPP8-KO_rep2_SRR5287123.txt
   # 60656 Input_H3K9me2or3_WT_rep1_SRR5287120.txt
   # 60656 Input_H3K9me2or3_WT_rep2_SRR5287121.txt
   # 60656 Input_H3K9me3_MORC2-KO_rep1_SRR5287108.txt
   # 60656 Input_H3K9me3_MORC2-KO_rep2_SRR5287109.txt
   # 60656 Input_H3K9me3_TASOR-KO_rep1_SRR5287110.txt
   # 60656 Input_H3K9me3_TASOR-KO_rep2_SRR5287111.txt
   # 60656 Input_H3K9me3_WT_rep1_SRR5287106.txt
   # 60656 Input_H3K9me3_WT_rep2_SRR5287107.txt
   # 60656 Input_MORC2_MORC2-KO_rep1_SRR5287094.txt
   # 60656 Input_MORC2_MORC2-KO_rep2_SRR5287095.txt
   # 60656 Input_MORC2_MPP8-KO_rep1_SRR5287096.txt
   # 60656 Input_MORC2_MPP8-KO_rep2_SRR5287097.txt
   # 60656 Input_MORC2_TASOR-KO_rep1_SRR5287098.txt
   # 60656 Input_MORC2_TASOR-KO_rep2_SRR5287099.txt
   # 60656 Input_MORC2_WT_rep1_SRR5287092.txt
   # 60656 Input_MORC2_WT_rep2_SRR5287093.txt


grep "ENSG00000180660.8" ChIP_H3K9me3_*.txt
# ChIP_H3K9me3_MORC2-KO_rep1_SRR5287102.txt:ENSG00000180660.8	32
# ChIP_H3K9me3_MORC2-KO_rep2_SRR5287103.txt:ENSG00000180660.8	52
# ChIP_H3K9me3_MPP8-KO_rep1_SRR5287118.txt:ENSG00000180660.8	50
# ChIP_H3K9me3_MPP8-KO_rep2_SRR5287119.txt:ENSG00000180660.8	60
# ChIP_H3K9me3_TASOR-KO_rep1_SRR5287104.txt:ENSG00000180660.8	55
# ChIP_H3K9me3_TASOR-KO_rep2_SRR5287105.txt:ENSG00000180660.8	59
# ChIP_H3K9me3_WT_rep1_SRR5287100.txt:ENSG00000180660.8	219
# ChIP_H3K9me3_WT_rep2_SRR5287101.txt:ENSG00000180660.8	276
# ChIP_H3K9me3_WT_rep3_SRR5287116.txt:ENSG00000180660.8	254
# ChIP_H3K9me3_WT_rep4_SRR5287117.txt:ENSG00000180660.8	192
```





## Tables

### Gene info + RNA-seq + ChIP-seq H3K9m3 + ChIP-seq MORC2 + ChIP-seq MPP8 + ChIP-seq TASOR (20210511)

```r
#module load R-core/3.6.0-foss-2017a
#export R_LIBS_USER=/projects/qbio/bifo/R/3.6.0
#cd /projects/qbio/bifo/users/sergio/repository/20201122_hush/table
#R

library(data.table)
library(GenomicFeatures)
library(edgeR)

# Enlarge the view width when printing tables
options(width = 250)

##############
# genes info #
##############
genes <- fread("/projects/qbio/bifo/users/sergio/repository/20201122_hush/annotation/gencode.v35.annotation.gene_type.intronless.bed")
setnames(genes, c("chr", "start", "end", "gene_name", "gene_id", "strand", "gene_type", "intronless"))

##############
# gene sizes #
##############
txdb <- makeTxDbFromGFF("/projects/qbio/bifo/users/sergio/reference/gencode/release35_GRCh38.p13/gtf/gencode.v35.annotation.gtf", format="gtf")
genes_length <- data.table(data.frame(genes(txdb)))[, c("gene_id", "width")]
genes_length[, length := width]
genes_length[, width := NULL]

genes <- merge(genes, genes_length, by = "gene_id")

##########
# rnaseq #
##########
rnaseq <- fread("tableCat.py -i /projects/qbio/bifo/users/sergio/repository/20201122_hush/rnaseq/multicov/*.txt -r .txt")
setnames(rnaseq, c("gene_id", "count", "file_id"))
rnaseq[file_id=="MORC2-KO_rep1_SRR5287126", file_id := "MORC2KO_rep1"]
rnaseq[file_id=="MORC2-KO_rep2_SRR5287127", file_id := "MORC2KO_rep2"]
rnaseq[file_id=="MPP8-KO_rep1_SRR5287128", file_id := "MPP8KO_rep1"]
rnaseq[file_id=="MPP8-KO_rep2_SRR5287129", file_id := "MPP8KO_rep2"]
rnaseq[file_id=="TASOR-KO_rep1_SRR5287130", file_id := "TASORKO_rep1"]
rnaseq[file_id=="TASOR-KO_rep2_SRR5287131", file_id := "TASORKO_rep2"]
rnaseq[file_id=="WT_rep1_SRR5287124", file_id := "WT_rep1"]
rnaseq[file_id=="WT_rep2_SRR5287125", file_id := "WT_rep2"]
rnaseq_dcast <- dcast(rnaseq, gene_id ~ file_id, value.var = "count")

# Define group
group <- factor(c('MORC2KO', 'MORC2KO', 'MPP8KO', 'MPP8KO', 'TASORKO', 'TASORKO', 'WT', 'WT'))

# Check that gene order is similar
sum(rnaseq_dcast$gene_id == genes$gene_id)

# Define DGEList object
y <- DGEList(counts = rnaseq_dcast[,c(-1)], group = group, genes = genes)

# Filtering
isexpr <- filterByExpr(y, min.count = 5)
table(isexpr)
#FALSE  TRUE
#27678 32978

y <- y[isexpr, , keep.lib.sizes=FALSE]
dim(y)
#32978     8

# Normalization
y <- calcNormFactors(y)
y$samples

# MDS
pdf("../rnaseq/figures/rnaseq_mds.pdf")
p <- plotMDS(y, col=rep(1:4, each=2))
dev.off()

# Design matrix
design <- model.matrix(~ 0 + as.vector(y$samples$group))
colnames(design) <- unique(as.vector(y$samples$group))

# Dispersion
y <- estimateDisp(y, design, robust=TRUE)
y$common.dispersion

# BCV
pdf("../rnaseq/figures/rnaseq_bcv.pdf")
p <- plotBCV(y)
dev.off()

# Fit genewise glms
fit <- glmFit(y, design)

# Differential expression
my.contrasts <- makeContrasts(MORC2KOvsWT = MORC2KO - WT, MPP8KOvsWT = MPP8KO - WT, TASORKOvsWT = TASORKO - WT, levels = design)

# Likelihood ratio tests
## MORC2KOvsWT
lrt_MORC2KOvsWT <- glmLRT(fit, contrast=my.contrasts[,"MORC2KOvsWT"])
table_MORC2KOvsWT <- data.table(data.frame(topTags(lrt_MORC2KOvsWT, n=Inf)))[, c("gene_id", "chr", "start", "end", "gene_name", "strand", "gene_type", "intronless", "length", "logFC", "FDR")]
## MPP8KOvsWT
lrt_MPP8KOvsWT <- glmLRT(fit, contrast=my.contrasts[,"MPP8KOvsWT"])
table_MPP8KOvsWT <- data.table(data.frame(topTags(lrt_MPP8KOvsWT, n=Inf)))[, c("gene_id", "chr", "start", "end", "gene_name", "strand", "gene_type", "intronless", "length", "logFC", "FDR")]
## TASORKOvsWT
lrt_TASORKOvsWT <- glmLRT(fit, contrast=my.contrasts[,"TASORKOvsWT"])
table_TASORKOvsWT <- data.table(data.frame(topTags(lrt_TASORKOvsWT, n=Inf)))[, c("gene_id", "chr", "start", "end", "gene_name", "strand", "gene_type", "intronless", "length", "logFC", "FDR")]

# Combine logFC and FDR and add RPKM
table_rnaseq_tmp <- merge(table_MORC2KOvsWT, table_MPP8KOvsWT, by = c("gene_id", "chr", "start", "end", "gene_name", "strand", "gene_type", "intronless", "length"), suffixes = c("_MORC2KOvsWT", "_MPP8KOvsWT"))
table_rnaseq <- merge(table_rnaseq_tmp, table_TASORKOvsWT, by = c("gene_id", "chr", "start", "end", "gene_name", "strand", "gene_type", "intronless", "length"))
table_rnaseq <- cbind(table_rnaseq, rpkm(y))
setnames(table_rnaseq, c("gene_id", "chr", "start", "end", "gene_name", "strand", "gene_type", "intronless", "length", "rnaseq_logFC_MORC2KOvsWT", "rnaseq_FDR_MORC2KOvsWT", "rnaseq_logFC_MPP8KOvsWT", "rnaseq_FDR_MPP8KOvsWT", "rnaseq_logFC_TASORKOvsWT", "rnaseq_FDR_TASORKOvsWT", "rnaseq_RPKM_MORC2KO_rep1", "rnaseq_RPKM_MORC2KO_rep2", "rnaseq_RPKM_MPP8KO_rep1", "rnaseq_RPKM_MPP8KO_rep2", "rnaseq_RPKM_TASORKO_rep1", "rnaseq_RPKM_TASORKO_rep2", "rnaseq_RPKM_WT_rep1", "rnaseq_RPKM_WT_rep2"))
table_rnaseq <- table_rnaseq[, c("gene_id", "chr", "start", "end", "gene_name", "strand", "gene_type", "intronless", "length", "rnaseq_RPKM_MORC2KO_rep1", "rnaseq_RPKM_MORC2KO_rep2", "rnaseq_RPKM_MPP8KO_rep1", "rnaseq_RPKM_MPP8KO_rep2", "rnaseq_RPKM_TASORKO_rep1", "rnaseq_RPKM_TASORKO_rep2", "rnaseq_RPKM_WT_rep1", "rnaseq_RPKM_WT_rep2", "rnaseq_logFC_MORC2KOvsWT", "rnaseq_FDR_MORC2KOvsWT", "rnaseq_logFC_MPP8KOvsWT", "rnaseq_FDR_MPP8KOvsWT", "rnaseq_logFC_TASORKOvsWT", "rnaseq_FDR_TASORKOvsWT")]



###################
# chipseq h3k9me3 #
###################
chipseq_h3k9me3 <- fread("tableCat.py -i /projects/qbio/bifo/users/sergio/repository/20201122_hush/chipseq/multicov/ChIP_H3K9me3*.txt -r .txt")
setnames(chipseq_h3k9me3, c("gene_id", "count", "file_id"))
chipseq_h3k9me3[file_id=="ChIP_H3K9me3_MORC2-KO_rep1_SRR5287102", file_id := "MORC2KO_rep1"]
chipseq_h3k9me3[file_id=="ChIP_H3K9me3_MORC2-KO_rep2_SRR5287103", file_id := "MORC2KO_rep2"]
chipseq_h3k9me3[file_id=="ChIP_H3K9me3_MPP8-KO_rep1_SRR5287118", file_id := "MPP8KO_rep1"]
chipseq_h3k9me3[file_id=="ChIP_H3K9me3_MPP8-KO_rep2_SRR5287119", file_id := "MPP8KO_rep2"]
chipseq_h3k9me3[file_id=="ChIP_H3K9me3_TASOR-KO_rep1_SRR5287104", file_id := "TASORKO_rep1"]
chipseq_h3k9me3[file_id=="ChIP_H3K9me3_TASOR-KO_rep2_SRR5287105", file_id := "TASORKO_rep2"]
chipseq_h3k9me3[file_id=="ChIP_H3K9me3_WT_rep1_SRR5287100", file_id := "WT_rep1"]
chipseq_h3k9me3[file_id=="ChIP_H3K9me3_WT_rep2_SRR5287101", file_id := "WT_rep2"]
chipseq_h3k9me3 <- chipseq_h3k9me3[file_id!="ChIP_H3K9me3_WT_rep3_SRR5287116" & file_id!="ChIP_H3K9me3_WT_rep4_SRR5287117"]
chipseq_h3k9me3_dcast <- dcast(chipseq_h3k9me3, gene_id ~ file_id, value.var = "count")

# Define group
group <- factor(c('MORC2KO', 'MORC2KO', 'MPP8KO', 'MPP8KO', 'TASORKO', 'TASORKO', 'WT', 'WT'))

# Check that gene order is similar
sum(chipseq_h3k9me3_dcast$gene_id == genes$gene_id)

# Define DGEList object
y <- DGEList(counts = chipseq_h3k9me3_dcast[,c(-1)], group = group, genes = genes)

# Filtering
isexpr <- filterByExpr(y, min.count = 5)
table(isexpr)
#FALSE  TRUE
# 6977 53679

y <- y[isexpr, , keep.lib.sizes=FALSE]
dim(y)
#53679     8

# Normalization
y <- calcNormFactors(y)
y$samples

# MDS
pdf("../chipseq/figures/chipseq_h3k9me3_mds.pdf")
p <- plotMDS(y, col=rep(1:4, each=2))
dev.off()

# Design matrix
design <- model.matrix(~ 0 + as.vector(y$samples$group))
colnames(design) <- unique(as.vector(y$samples$group))

# Dispersion
y <- estimateDisp(y, design, robust=TRUE)
y$common.dispersion

# BCV
pdf("../chipseq/figures/chipseq_h3k9me3_bcv.pdf")
p <- plotBCV(y)
dev.off()

# Fit genewise glms
fit <- glmFit(y, design)

# Differential expression
my.contrasts <- makeContrasts(MORC2KOvsWT = MORC2KO - WT, MPP8KOvsWT = MPP8KO - WT, TASORKOvsWT = TASORKO - WT, levels = design)

# Likelihood ratio tests
## MORC2KOvsWT
lrt_MORC2KOvsWT <- glmLRT(fit, contrast=my.contrasts[,"MORC2KOvsWT"])
table_MORC2KOvsWT <- data.table(data.frame(topTags(lrt_MORC2KOvsWT, n=Inf)))[, c("gene_id", "chr", "start", "end", "gene_name", "strand", "gene_type", "intronless", "length", "logFC", "FDR")]
## MPP8KOvsWT
lrt_MPP8KOvsWT <- glmLRT(fit, contrast=my.contrasts[,"MPP8KOvsWT"])
table_MPP8KOvsWT <- data.table(data.frame(topTags(lrt_MPP8KOvsWT, n=Inf)))[, c("gene_id", "chr", "start", "end", "gene_name", "strand", "gene_type", "intronless", "length", "logFC", "FDR")]
## TASORKOvsWT
lrt_TASORKOvsWT <- glmLRT(fit, contrast=my.contrasts[,"TASORKOvsWT"])
table_TASORKOvsWT <- data.table(data.frame(topTags(lrt_TASORKOvsWT, n=Inf)))[, c("gene_id", "chr", "start", "end", "gene_name", "strand", "gene_type", "intronless", "length", "logFC", "FDR")]

# Combine logFC and FDR and add RPKM
table_chipseq_h3k9me3_tmp <- merge(table_MORC2KOvsWT, table_MPP8KOvsWT, by = c("gene_id", "chr", "start", "end", "gene_name", "strand", "gene_type", "intronless", "length"), suffixes = c("_MORC2KOvsWT", "_MPP8KOvsWT"))
table_chipseq_h3k9me3 <- merge(table_chipseq_h3k9me3_tmp, table_TASORKOvsWT, by = c("gene_id", "chr", "start", "end", "gene_name", "strand", "gene_type", "intronless", "length"))
table_chipseq_h3k9me3 <- cbind(table_chipseq_h3k9me3, rpkm(y))
setnames(table_chipseq_h3k9me3, c("gene_id", "chr", "start", "end", "gene_name", "strand", "gene_type", "intronless", "length", "chipseqh3k9me3_logFC_MORC2KOvsWT", "chipseqh3k9me3_FDR_MORC2KOvsWT", "chipseqh3k9me3_logFC_MPP8KOvsWT", "chipseqh3k9me3_FDR_MPP8KOvsWT", "chipseqh3k9me3_logFC_TASORKOvsWT", "chipseqh3k9me3_FDR_TASORKOvsWT", "chipseqh3k9me3_RPKM_MORC2KO_rep1", "chipseqh3k9me3_RPKM_MORC2KO_rep2", "chipseqh3k9me3_RPKM_MPP8KO_rep1", "chipseqh3k9me3_RPKM_MPP8KO_rep2", "chipseqh3k9me3_RPKM_TASORKO_rep1", "chipseqh3k9me3_RPKM_TASORKO_rep2", "chipseqh3k9me3_RPKM_WT_rep1", "chipseqh3k9me3_RPKM_WT_rep2"))
table_chipseq_h3k9me3 <- table_chipseq_h3k9me3[, c("gene_id", "chr", "start", "end", "gene_name", "strand", "gene_type", "intronless", "length", "chipseqh3k9me3_RPKM_MORC2KO_rep1", "chipseqh3k9me3_RPKM_MORC2KO_rep2", "chipseqh3k9me3_RPKM_MPP8KO_rep1", "chipseqh3k9me3_RPKM_MPP8KO_rep2", "chipseqh3k9me3_RPKM_TASORKO_rep1", "chipseqh3k9me3_RPKM_TASORKO_rep2", "chipseqh3k9me3_RPKM_WT_rep1", "chipseqh3k9me3_RPKM_WT_rep2", "chipseqh3k9me3_logFC_MORC2KOvsWT", "chipseqh3k9me3_FDR_MORC2KOvsWT", "chipseqh3k9me3_logFC_MPP8KOvsWT", "chipseqh3k9me3_FDR_MPP8KOvsWT", "chipseqh3k9me3_logFC_TASORKOvsWT", "chipseqh3k9me3_FDR_TASORKOvsWT")]



#################
# chipseq morc2 #
#################
chipseq_morc2 <- fread("tableCat.py -i /projects/qbio/bifo/users/sergio/repository/20201122_hush/chipseq/multicov/ChIP_MORC2_{MORC2-KO,WT}*.txt -r .txt")
setnames(chipseq_morc2, c("gene_id", "count", "file_id"))
chipseq_morc2[file_id=="ChIP_MORC2_MORC2-KO_rep1_SRR5287070", file_id := "MORC2KO_rep1"]
chipseq_morc2[file_id=="ChIP_MORC2_MORC2-KO_rep2_SRR5287071", file_id := "MORC2KO_rep2"]
chipseq_morc2[file_id=="ChIP_MORC2_WT_rep1_SRR5287068", file_id := "WT_rep1"]
chipseq_morc2[file_id=="ChIP_MORC2_WT_rep2_SRR5287069", file_id := "WT_rep2"]
chipseq_morc2_dcast <- dcast(chipseq_morc2, gene_id ~ file_id, value.var = "count")

# Define group
group <- factor(c('MORC2KO', 'MORC2KO', 'WT', 'WT'))

# Check that gene order is similar
sum(chipseq_morc2_dcast$gene_id == genes$gene_id)

# Define DGEList object
y <- DGEList(counts = chipseq_morc2_dcast[,c(-1)], group = group, genes = genes)

# Filtering
isexpr <- filterByExpr(y, min.count = 5)
table(isexpr)
#FALSE  TRUE
#14457 46199

y <- y[isexpr, , keep.lib.sizes=FALSE]
dim(y)
#46199     4

# Normalization
y <- calcNormFactors(y)
y$samples

# MDS
pdf("../chipseq/figures/chipseq_morc2_mds.pdf")
p <- plotMDS(y, col=rep(1:2, each=2))
dev.off()

# Design matrix
design <- model.matrix(~ 0 + as.vector(y$samples$group))
colnames(design) <- unique(as.vector(y$samples$group))

# Dispersion
y <- estimateDisp(y, design, robust=TRUE)
y$common.dispersion

# BCV
pdf("../chipseq/figures/chipseq_morc2_bcv.pdf")
p <- plotBCV(y)
dev.off()

# Fit genewise glms
fit <- glmFit(y, design)

# Differential expression
my.contrasts <- makeContrasts(MORC2KOvsWT = MORC2KO - WT, levels = design)

# Likelihood ratio tests
## MORC2KOvsWT
lrt_MORC2KOvsWT <- glmLRT(fit, contrast=my.contrasts[,"MORC2KOvsWT"])
table_MORC2KOvsWT <- data.table(data.frame(topTags(lrt_MORC2KOvsWT, n=Inf)))[, c("gene_id", "chr", "start", "end", "gene_name", "strand", "gene_type", "intronless", "length", "logFC", "FDR")]

# Combine logFC and FDR and add RPKM
table_chipseq_morc2 <- cbind(table_MORC2KOvsWT, rpkm(y))
setnames(table_chipseq_morc2, c("gene_id", "chr", "start", "end", "gene_name", "strand", "gene_type", "intronless", "length", "chipseqmorc2_logFC_MORC2KOvsWT", "chipseqmorc2_FDR_MORC2KOvsWT", "chipseqmorc2_RPKM_MORC2KO_rep1", "chipseqmorc2_RPKM_MORC2KO_rep2", "chipseqmorc2_RPKM_WT_rep1", "chipseqmorc2_RPKM_WT_rep2"))
table_chipseq_morc2 <- table_chipseq_morc2[, c("gene_id", "chr", "start", "end", "gene_name", "strand", "gene_type", "intronless", "length", "chipseqmorc2_RPKM_MORC2KO_rep1", "chipseqmorc2_RPKM_MORC2KO_rep2", "chipseqmorc2_RPKM_WT_rep1", "chipseqmorc2_RPKM_WT_rep2", "chipseqmorc2_logFC_MORC2KOvsWT", "chipseqmorc2_FDR_MORC2KOvsWT")]



################
# chipseq mpp8 #
################
chipseq_mpp8 <- fread("tableCat.py -i /projects/qbio/bifo/users/sergio/repository/20201122_hush/chipseq/multicov/ChIP_MPP8_{MPP8-KO,WT}*.txt -r .txt")
setnames(chipseq_mpp8, c("gene_id", "count", "file_id"))
chipseq_mpp8[file_id=="ChIP_MPP8_MPP8-KO_rep1_SRR5287080", file_id := "MPP8KO_rep1"]
chipseq_mpp8[file_id=="ChIP_MPP8_MPP8-KO_rep2_SRR5287081", file_id := "MPP8KO_rep2"]
chipseq_mpp8[file_id=="ChIP_MPP8_WT_rep1_SRR5287076", file_id := "WT_rep1"]
chipseq_mpp8[file_id=="ChIP_MPP8_WT_rep2_SRR5287077", file_id := "WT_rep2"]
chipseq_mpp8_dcast <- dcast(chipseq_mpp8, gene_id ~ file_id, value.var = "count")

# Define group
group <- factor(c('MPP8KO', 'MPP8KO', 'WT', 'WT'))

# Check that gene order is similar
sum(chipseq_mpp8_dcast$gene_id == genes$gene_id)

# Define DGEList object
y <- DGEList(counts = chipseq_mpp8_dcast[,c(-1)], group = group, genes = genes)

# Filtering
isexpr <- filterByExpr(y, min.count = 5)
table(isexpr)
#FALSE  TRUE
#12732 47924

y <- y[isexpr, , keep.lib.sizes=FALSE]
dim(y)
#47924     4

# Normalization
y <- calcNormFactors(y)
y$samples

# MDS
pdf("../chipseq/figures/chipseq_mpp8_mds.pdf")
p <- plotMDS(y, col=rep(1:2, each=2))
dev.off()

# Design matrix
design <- model.matrix(~ 0 + as.vector(y$samples$group))
colnames(design) <- unique(as.vector(y$samples$group))

# Dispersion
y <- estimateDisp(y, design, robust=TRUE)
y$common.dispersion

# BCV
pdf("../chipseq/figures/chipseq_mpp8_bcv.pdf")
p <- plotBCV(y)
dev.off()

# Fit genewise glms
fit <- glmFit(y, design)

# Differential expression
my.contrasts <- makeContrasts(MPP8KOvsWT = MPP8KO - WT, levels = design)

# Likelihood ratio tests
## MPP8KOvsWT
lrt_MPP8KOvsWT <- glmLRT(fit, contrast=my.contrasts[,"MPP8KOvsWT"])
table_MPP8KOvsWT <- data.table(data.frame(topTags(lrt_MPP8KOvsWT, n=Inf)))[, c("gene_id", "chr", "start", "end", "gene_name", "strand", "gene_type", "intronless", "length", "logFC", "FDR")]

# Combine logFC and FDR and add RPKM
table_chipseq_mpp8 <- cbind(table_MPP8KOvsWT, rpkm(y))
setnames(table_chipseq_mpp8, c("gene_id", "chr", "start", "end", "gene_name", "strand", "gene_type", "intronless", "length", "chipseqmpp8_logFC_MPP8KOvsWT", "chipseqmpp8_FDR_MPP8KOvsWT", "chipseqmpp8_RPKM_MPP8KO_rep1", "chipseqmpp8_RPKM_MPP8KO_rep2", "chipseqmpp8_RPKM_WT_rep1", "chipseqmpp8_RPKM_WT_rep2"))
table_chipseq_mpp8 <- table_chipseq_mpp8[, c("gene_id", "chr", "start", "end", "gene_name", "strand", "gene_type", "intronless", "length", "chipseqmpp8_RPKM_MPP8KO_rep1", "chipseqmpp8_RPKM_MPP8KO_rep2", "chipseqmpp8_RPKM_WT_rep1", "chipseqmpp8_RPKM_WT_rep2", "chipseqmpp8_logFC_MPP8KOvsWT", "chipseqmpp8_FDR_MPP8KOvsWT")]



#################
# chipseq tasor #
#################
chipseq_tasor <- fread("tableCat.py -i /projects/qbio/bifo/users/sergio/repository/20201122_hush/chipseq/multicov/ChIP_TASOR_{TASOR-KO,WT}*.txt -r .txt")
setnames(chipseq_tasor, c("gene_id", "count", "file_id"))
chipseq_tasor[file_id=="ChIP_TASOR_TASOR-KO_rep1_SRR5287090", file_id := "TASORKO_rep1"]
chipseq_tasor[file_id=="ChIP_TASOR_TASOR-KO_rep2_SRR5287091", file_id := "TASORKO_rep2"]
chipseq_tasor[file_id=="ChIP_TASOR_WT_rep1_SRR5287084", file_id := "WT_rep1"]
chipseq_tasor[file_id=="ChIP_TASOR_WT_rep2_SRR5287085", file_id := "WT_rep2"]
chipseq_tasor_dcast <- dcast(chipseq_tasor, gene_id ~ file_id, value.var = "count")

# Define group
group <- factor(c('TASORKO', 'TASORKO', 'WT', 'WT'))

# Check that gene order is similar
sum(chipseq_tasor_dcast$gene_id == genes$gene_id)

# Define DGEList object
y <- DGEList(counts = chipseq_tasor_dcast[,c(-1)], group = group, genes = genes)

# Filtering
isexpr <- filterByExpr(y, min.count = 5)
table(isexpr)
#FALSE  TRUE
#13164 47492

y <- y[isexpr, , keep.lib.sizes=FALSE]
dim(y)
#47492     4

# Normalization
y <- calcNormFactors(y)
y$samples

# MDS
pdf("../chipseq/figures/chipseq_tasor_mds.pdf")
p <- plotMDS(y, col=rep(1:2, each=2))
dev.off()

# Design matrix
design <- model.matrix(~ 0 + as.vector(y$samples$group))
colnames(design) <- unique(as.vector(y$samples$group))

# Dispersion
y <- estimateDisp(y, design, robust=TRUE)
y$common.dispersion

# BCV
pdf("../chipseq/figures/chipseq_tasor_bcv.pdf")
p <- plotBCV(y)
dev.off()

# Fit genewise glms
fit <- glmFit(y, design)

# Differential expression
my.contrasts <- makeContrasts(TASORKOvsWT = TASORKO - WT, levels = design)

# Likelihood ratio tests
## TASORKOvsWT
lrt_TASORKOvsWT <- glmLRT(fit, contrast=my.contrasts[,"TASORKOvsWT"])
table_TASORKOvsWT <- data.table(data.frame(topTags(lrt_TASORKOvsWT, n=Inf)))[, c("gene_id", "chr", "start", "end", "gene_name", "strand", "gene_type", "intronless", "length", "logFC", "FDR")]

# Combine logFC and FDR and add RPKM
table_chipseq_tasor <- cbind(table_TASORKOvsWT, rpkm(y))
setnames(table_chipseq_tasor, c("gene_id", "chr", "start", "end", "gene_name", "strand", "gene_type", "intronless", "length", "chipseqtasor_logFC_TASORKOvsWT", "chipseqtasor_FDR_TASORKOvsWT", "chipseqtasor_RPKM_TASORKO_rep1", "chipseqtasor_RPKM_TASORKO_rep2", "chipseqtasor_RPKM_WT_rep1", "chipseqtasor_RPKM_WT_rep2"))
table_chipseq_tasor <- table_chipseq_tasor[, c("gene_id", "chr", "start", "end", "gene_name", "strand", "gene_type", "intronless", "length", "chipseqtasor_RPKM_TASORKO_rep1", "chipseqtasor_RPKM_TASORKO_rep2", "chipseqtasor_RPKM_WT_rep1", "chipseqtasor_RPKM_WT_rep2", "chipseqtasor_logFC_TASORKOvsWT", "chipseqtasor_FDR_TASORKOvsWT")]



#########################
# merge and write table #
#########################
genes_rnaseq <- merge(genes, table_rnaseq, by = c("gene_id", "chr", "start", "end", "gene_name", "strand", "gene_type", "intronless", "length"), all.x = TRUE)
genes_rnaseq_chipseqh3k9me3 <- merge(genes_rnaseq, table_chipseq_h3k9me3, by = c("gene_id", "chr", "start", "end", "gene_name", "strand", "gene_type", "intronless", "length"), all.x = TRUE)
genes_rnaseq_chipseqh3k9me3_chipseqmorc2 <- merge(genes_rnaseq_chipseqh3k9me3, table_chipseq_morc2, by = c("gene_id", "chr", "start", "end", "gene_name", "strand", "gene_type", "intronless", "length"), all.x = TRUE)
genes_rnaseq_chipseqh3k9me3_chipseqmorc2_chipseqmpp8 <- merge(genes_rnaseq_chipseqh3k9me3_chipseqmorc2, table_chipseq_mpp8, by = c("gene_id", "chr", "start", "end", "gene_name", "strand", "gene_type", "intronless", "length"), all.x = TRUE)
genes_rnaseq_chipseqh3k9me3_chipseqmorc2_chipseqmpp8_chipseqtasor <- merge(genes_rnaseq_chipseqh3k9me3_chipseqmorc2_chipseqmpp8, table_chipseq_tasor, by = c("gene_id", "chr", "start", "end", "gene_name", "strand", "gene_type", "intronless", "length"), all.x = TRUE)

write.table(genes_rnaseq_chipseqh3k9me3_chipseqmorc2_chipseqmpp8_chipseqtasor, "20210511_table.txt", sep="\t", quote = FALSE, row.names = FALSE, col.names = TRUE)
```
